### DISCUSSION TITLE
Corsair Controller for Corsair Link

### DISCUSSION THREAD

**Discussion by scole250** [liquidctl/liquidctl#658] [General]
I have a system with a 7-8 yr old Corsair Controller for Corsair Link USB device. Under Windows, Corsair Commander sees it and can control all the Corsair devices including a H100i and AX1200i and it sees the Corsair (non-RGB) Vengeance DIMMs. I also run that same system under Linux Mint 20.3 Xfce. Running under linux, I've installed Liquidctl but when I run liquidctl list it doesn't show any devices, no error messages, outputs nothing. When I run lsusb, I get this...
Bus 002 Device 003: ID 1b1c:1c00 Corsair Controller for Corsair Link
Bus 002 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 002 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 008 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 007 Device 002: ID 1b1c:0c04 Corsair Link Cooling Node
Bus 007 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 006 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 005 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 001 Device 002: ID 8087:0024 Intel Corp. Integrated Rate Matching Hub
Bus 001 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub
Bus 004 Device 001: ID 1d6b:0003 Linux Foundation 3.0 root hub
Bus 003 Device 001: ID 1d6b:0002 Linux Foundation 2.0 root hub

Of course, when I run liquidctl initialize all, I get...
ERROR: no device matches available drivers and selection criteria

Any help would be appreciated!

**Comment by aleksamagicka** [comment]
I guess Corsair naming of devices strikes again, since Google turns no meaningful results for `Corsair Controller for Corsair Link`. It sounds like a hub for other Corsair devices, correct?

Either way, neither `1b1c:1c00` nor `1b1c:0c04` are explicitly listed in liquidctl right now. There seems to be some level of support in [OpenCorsairLink](https://github.com/audiohacked/OpenCorsairLink), though I don't know how much. There's a [guide](https://github.com/liquidctl/liquidctl/blob/main/docs/developer/porting-drivers-from-opencorsairlink.md) on porting device support from there to liquidctl.
